Build failure (cross-compiling): `usb-modeswitch` uses wrong `pkg-config`

### Steps To Reproduce

Steps to reproduce the behavior:
* I'm trying to build an `aarch64-linux` host from `x86_64-linux`
* I've historically disabled the `hardware.usb-modeswitch` option whenever cross-compiling

However, it seems like a relatively simple program, merely limited by the good ole bare Makefile. And my aarch64 sbc would benefit from a wifi adapter that needs this module/package.

I was going to dive in and try to just make it a meson project, but then I realized we already carry a patch for the Makefile, so I figured I'd open this before I forget, in case I don't have time to dig in:

### Build log

```
usb-modeswitch-aarch64-unknown-linux-gnu> Running phase: unpackPhase
usb-modeswitch-aarch64-unknown-linux-gnu> unpacking source archive /nix/store/m7vas4na0b0hbcf085mh7jpm978xfmgj-usb-modeswitch-2.6.0.tar.bz2
usb-modeswitch-aarch64-unknown-linux-gnu> source root is usb-modeswitch-2.6.0
usb-modeswitch-aarch64-unknown-linux-gnu> setting SOURCE_DATE_EPOCH to timestamp 1574975208 of file usb-modeswitch-2.6.0/make_string.sh
usb-modeswitch-aarch64-unknown-linux-gnu> Running phase: patchPhase
usb-modeswitch-aarch64-unknown-linux-gnu> applying patch /nix/store/fihsrgy4vak1xj37lcxlqkmijy4vyasq-configurable-usb-modeswitch.patch
usb-modeswitch-aarch64-unknown-linux-gnu> patching file Makefile
usb-modeswitch-aarch64-unknown-linux-gnu> patching file usb-modeswitch-upstart.conf
usb-modeswitch-aarch64-unknown-linux-gnu> patching file usb_modeswitch.sh
usb-modeswitch-aarch64-unknown-linux-gnu> patching file usb_modeswitch@.service
usb-modeswitch-aarch64-unknown-linux-gnu> patching file usb_modeswitch_dispatcher.tcl
usb-modeswitch-aarch64-unknown-linux-gnu> Running phase: updateAutotoolsGnuConfigScriptsPhase
usb-modeswitch-aarch64-unknown-linux-gnu> Running phase: updateAutotoolsGnuConfigScriptsPhase
usb-modeswitch-aarch64-unknown-linux-gnu> Running phase: configurePhase
usb-modeswitch-aarch64-unknown-linux-gnu> no configure script, doing nothing
usb-modeswitch-aarch64-unknown-linux-gnu> Running phase: buildPhase
usb-modeswitch-aarch64-unknown-linux-gnu> build flags: SHELL=/nix/store/9vafkkic27k7m4934fpawl6yip3a6k4h-bash-5.2-p21/bin/bash PREFIX=\$\(out\) ETCDIR=/etc USE_UPSTART=false USE_SYSTEMD=true SYSDIR=\$\(out\)/lib/systemd/system UDEVDIR=\$\(out\)/lib/udev
usb-modeswitch-aarch64-unknown-linux-gnu> aarch64-unknown-linux-gnu-gcc -o usb_modeswitch usb_modeswitch.c -Wall -Wno-deprecated-declarations `pkg-config --libs --cflags libusb-1.0`
usb-modeswitch-aarch64-unknown-linux-gnu> /nix/store/9vafkkic27k7m4934fpawl6yip3a6k4h-bash-5.2-p21/bin/bash: line 1: pkg-config: command not found
usb-modeswitch-aarch64-unknown-linux-gnu> In file included from usb_modeswitch.c:59:
usb-modeswitch-aarch64-unknown-linux-gnu> usb_modeswitch.h:26:10: fatal error: libusb.h: No such file or directory
usb-modeswitch-aarch64-unknown-linux-gnu>    26 | #include <libusb.h>
usb-modeswitch-aarch64-unknown-linux-gnu>       |          ^~~~~~~~~~
usb-modeswitch-aarch64-unknown-linux-gnu> compilation terminated.
usb-modeswitch-aarch64-unknown-linux-gnu> make: *** [Makefile:36: usb_modeswitch] Error 1
```

### Additional context

Add any other context about the problem here.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

This is on the latest usb-modeswitch rev from 2019, from `nixos-unstable`: 612f97239e2cc474c13c9dafa0df378058c5ad8d

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/tools/misc/usb-modeswitch/default.nix
+++ b/pkgs/development/tools/misc/usb-modeswitch/default.nix
@@ -10,7 +10,10 @@ stdenv.mkDerivation rec {
     sha256 = "18wbbxc5cfsmikba0msdvd5qlaga27b32nhrzicyd9mdddp265f2";
   };
 
-  patches = [ ./configurable-usb-modeswitch.patch ];
+  patches = [
+    ./configurable-usb-modeswitch.patch
+    ./pkg-config.patch
+  ];
 
   # Remove attempts to write to /etc and /var/lib.
   postPatch = ''

--- a/pkgs/development/tools/misc/usb-modeswitch/pkg-config.patch
+++ b/pkgs/development/tools/misc/usb-modeswitch/pkg-config.patch
@@ -1,0 +1,14 @@
+diff --git a/Makefile b/Makefile
+index 9aa7356..c5f2a87 100644
+--- a/Makefile
++++ b/Makefile
+@@ -2,7 +2,8 @@ PROG        = usb_modeswitch
+ VERS        = 2.6.0
+ CC          ?= gcc
+ CFLAGS      += -Wall -Wno-deprecated-declarations
+-LIBS        = `pkg-config --libs --cflags libusb-1.0`
++PKG_CONFIG  ?= pkg-config
++LIBS        = `$(PKG_CONFIG) --libs --cflags libusb-1.0`
+ RM          = /bin/rm -f
+ OBJS        = usb_modeswitch.c
+ PREFIX      = $(DESTDIR)/usr

